sklearn/tree/_criterion.pyx

```diff
@@ -230,10 +261,19 @@ cdef class ClassificationCriterion(Criterion):
         self.start = 0
         self.pos = 0
         self.end = 0
+        self.missing_direction = 2  # Ignore the missing values by default
 
         self.n_outputs = n_outputs
+
+        self.n_missing = 0
+        self.weighted_n_missing = 0
+
+        self.n_nonmissing = 0
+        self.weighted_n_nonmissing = 0
```

Comment:
`weighted_n_not_missing`? Or could I avoid this altogether and compute that value on the fly (It's required only at 3-4 places)

Changing `weighted_n_node_samples` to have the weighted sum for just the ones which are not missing would require change in the impurity computation, which right now remains unaltered.